tests/src/JIT/Performance/CodeQuality/HWIntrinsic/X86/PacketTracer/PacketTracer.cs

```diff
@@ -0,0 +1,253 @@
+            for (int x = 0; x < Width; x += VectorPacket256.Packet256Size)
+            {
+                float fx = (float)x;
+                Vector256<float> Xs = Add(SetAllVector256(fx), SevenToZero);
+                var dirs = GetPoints(Xs, SetAllVector256<float>(y), camera);
+                var rayPacket256 = new RayPacket256(camera.Pos, dirs);
+                var SoAcolors = TraceRay(rayPacket256, scene, 0);
+ 
+                var AoS = SoAcolors.FastTranspose();
+                var intAoS = AoS.ConvertToIntRGB();
+
+                int* output = &rgb[(x + stride) * 3];
+                {
+                    Store(output, GetLowerHalf<int>(intAoS.Rs));
+                    Store(output + 4, GetLowerHalf<int>(intAoS.Gs));
+                    Store(output + 8, GetLowerHalf<int>(intAoS.Bs));
+                    Avx2.ExtractVector128(output + 12, intAoS.Rs, 1);
+                    Avx2.ExtractVector128(output + 16, intAoS.Gs, 1);
+                    Avx2.ExtractVector128(output + 20, intAoS.Bs, 1);
+                }
+                /* 
+                var AoS = SoAcolors.Transpose();
+                var intAoS = AoS.ConvertToIntRGB();
+
+                int* output = &rgb[(x + stride) * 3];
+                {
+                    Store(output, intAoS.Rs);
+                    Store(output + 8, intAoS.Gs);
+                    Store(output + 16, intAoS.Bs);
+                }
+                */
+            }
+        }
+
+    }
+
+    private ColorPacket256 TraceRay(RayPacket256 rayPacket256, Scene scene, int depth)
+    {
+        var isect = MinIntersections(rayPacket256, scene);
+        if (isect.AllNullIntersections())
+        {
+            return ColorPacket256Helper.BackgroundColor;
+        }
+        var color = Shade(isect, rayPacket256, scene, depth);
+        var isNull = Compare(isect.Distances, Intersections.NullDistance, FloatComparisonMode.EqualOrderedNonSignaling);
+        var backgroundColor = ColorPacket256Helper.BackgroundColor.Xs;
+        return new ColorPacket256(BlendVariable(color.Xs, backgroundColor, isNull),
+                                  BlendVariable(color.Ys, backgroundColor, isNull),
+                                  BlendVariable(color.Zs, backgroundColor, isNull));
+    }
+
+    private Vector256<float> TestRay(RayPacket256 rayPacket256, Scene scene)
+    {
+        var isect = MinIntersections(rayPacket256, scene);
+        if (isect.AllNullIntersections())
+        {
+            return SetZeroVector256<float>();
+        }
+        var isNull = Compare(isect.Distances, Intersections.NullDistance, FloatComparisonMode.EqualOrderedNonSignaling);
+        return BlendVariable(isect.Distances, SetZeroVector256<float>(), isNull);
+    }
+
+    private Intersections MinIntersections(RayPacket256 rayPacket256, Scene scene)
+    {
+        Intersections mins = new Intersections(Intersections.NullDistance, Intersections.NullIndex);
+        for (int i = 0; i < scene.Things.Length; i++)
```

Comment:
I think its better to cache `Length` in a local...

Reply:
(I might be miss-remembering, however)

Reply:
Thanks, will fix.